### PR TITLE
feat(api-client): request sidebar dragging

### PR DESCRIPTION
**Pull request by antlio**
this pr is a proposals to support sidebar dragging while adding some transition sugar:

[Uploading scalar-sidebar.mp4…](https://github.com/scalar/scalar/assets/14966155/feae7a15-e9b9-4836-80de-870afa235d07)

### CODE REVIEW

**Comment by cameronrohani** [comment]
Not sure if this makes sense to pursue _right in this moment_ but there's some things that would ideally happen if we were to add this:

- Sidebar size doesn't save on page refresh / collapse
https://github.com/user-attachments/assets/7093cfed-81b6-4208-ad94-e8dd99bbd012

- Sidebar icon flickers when hide / show state 
https://github.com/user-attachments/assets/1fce654f-574d-46b1-bbaf-0cc51b4be283

Also If we add this here should we be adding it in other places i.e request / response?

**Comment by antlio** [comment]
@cameronrohani 
> Sidebar size doesn't save on page refresh / collapse

added

> Sidebar icon flickers when hide / show state

this is not related to that PR, however we should def fix it!

> Also If we add this here should we be adding it in other places i.e request / response?

we could if its worth it!

File: packages/api-client/src/components/Sidebar/Sidebar.vue [hunk 1]
```diff
@@ -128,10 +128,6 @@ watch(
   height: 100%;
   z-index: 99;
 }
-.sidebar-enter-active,
-.sidebar-leave-active {
-  transition: width 0.15s ease !important;
-}
 .sidebar-enter-from,
 .sidebar-leave-to {
   width: 0 !important;
```
Comment (antlio) [inline packages/api-client/src/components/Sidebar/Sidebar.vue:143]: does this feel too harsh without any transition?

File: packages/api-client/src/components/Sidebar/Sidebar.vue [hunk 2]
```diff
@@ -158,7 +154,5 @@ watch(
   width: 100%;
   height: 100%;
   z-index: 100;
-  backdrop-filter: blur(1.5px);
-  transition: backdrop-filter 0.15s ease-in-out;
```
Comment (antlio) [inline packages/api-client/src/components/Sidebar/Sidebar.vue]: before element is to be removed if we don't rely on backdrop filter / transition

File: packages/api-client/src/components/Sidebar/Sidebar.vue [hunk 3]
```diff
@@ -1,23 +1,179 @@
...
 const { isReadOnly } = useWorkspace()
+const isDragging = ref(false)
+const sidebarRef = ref<HTMLElement | null>(null)
+const sidebarWidth = ref(localStorage.getItem('sidebarWidth') || '280px')
+
+/**
+ * Resets the sidebar width to the default value to prevent
+ * having it shrinked on click display after dragging.
+ */
+const initialSidebarWidth = () => {
```
Comment (geoffgscott) [inline packages/api-client/src/components/Sidebar/Sidebar.vue:20]: Nitpick but would prefer a function declaration here and an action word for function naming such as:

```ts
function setInitialSidebarWidth()
```

**Comment by marclave** [comment]
awesome pr @antlio ! i think there are some second order effects here that we need to think about in regards to resizing everything other sections, but also let's revisit post invite only release of the app ✨

**Comment by hanspagel** [comment]
What if we get this in, but make sure it’s disabled/not active yet? me doesn’t like stale PRs @marclave @antlio

**Comment by antlio** [comment]
> What if we get this in, but make sure it’s disabled/not active yet? me doesn’t like stale PRs @marclave @antlio

i cleaned/rebased it in case - will let @marclave make the last move (close/merge)

**Comment by marclave** [comment]
let's get it in :) its a good idea, we can think second order effects later <3 great work @antlio

**Comment by cameronrohani** [comment]
Awesome excited to get this in

I think the main side effect that we'll need to fix is having the references in the client will need to persist the sidebar size ideally...

I think we should keep the transition for close (did I get rid of it? I can try to add it back)

**Comment by cameronrohani** [comment]
after marc and I played around with it for a bit we decided to remove all transitions, ideally in the closed state it would keep the additional border if ur cursor is still clicked down (see vscode)

**Pull request merged**